### ISSUE TITLE
Bug: Facebook Sdk Initialization Error

### ISSUE THREAD

**Issue by talhaozdemir** [pixijs/pixijs#10086]
### Current Behavior

When try to add script like

```
<script>
  window.fbAsyncInit = function() {
    FB.init({
      appId            : 'your-app-id',
      xfbml            : true,
      version          : 'v18.0'
    });
  };
</script>
<script async defer crossorigin="anonymous" src="https://connect.facebook.net/en_US/sdk.js"></script>
```

I got error:

Uncaught (in promise) TypeError: Path must be a string. Received undefined 

from @pixi/utils/lib/path.mjs

### Expected Behavior

I don't think that this is because Facebook API, because this API is used commonly over the world.

### Steps to Reproduce

Just do instruction from the [link](https://developers.facebook.com/docs/javascript/quickstart/)

If I download this [script](https://connect.facebook.net/en_US/sdk.js) and import it manually I got same error, why pixi throws error for an external script?

### Environment

- **`pixi.js` version**: 7.2.4*
- **Browser & Version**: Chrome, Firefox
- **OS & Version**: Windows 11

**Comment by SuperSodaSea** [comment]
Can you show how you import and use PixiJS? It would be nice if you could provide a full stack trace of the errors.

**Comment by talhaozdemir** [comment]
```
//assets.ts

import manifest from "./manifest.json";
import { Assets } from "pixi.js";

.
.
.
export async function initAssets() {
await Assets.init({ manifest });
const assets = await Assets.loadBundle(["fonts", "default", "sounds", "shop"]);
.
.
.
}
```

![image](https://github.com/pixijs/pixijs/assets/37840852/121c804a-f3b0-40b9-a640-decfa7e096a1)

**Comment by talhaozdemir** [comment]
I logged the paths, they may help you.

![image](https://github.com/pixijs/pixijs/assets/37840852/3573a599-e3db-469b-a4e9-0d1adbae486f)

![image](https://github.com/pixijs/pixijs/assets/37840852/18428557-79b1-426e-8a74-8ceb39a34669)

**Comment by talhaozdemir** [comment]
If I remove Facebook SDK from index.html, undefined consoles disappear that above picture.

**Comment by bigtimebuddy** [comment]
Maybe try debugging at the `promises` call in the stack. See what the `assetsToLoad` looks like.

**Comment by talhaozdemir** [comment]
Here it is;

with Facebook SDK
![image](https://github.com/pixijs/pixijs/assets/37840852/77ee34bb-c5b0-4386-baea-79687aca1d40)
without
![image](https://github.com/pixijs/pixijs/assets/37840852/d8942026-7f91-4a2a-aee8-5abe3f6a3635)

**Comment by talhaozdemir** [comment]
I also need to mention that sometimes it doesn't give the error in Chrome, and the application opens, but it always gives the error in Firefox.

**Comment by SuperSodaSea** [comment]
Looks like the Facebook SDK injected the array somewhere :thinking:

**Comment by bigtimebuddy** [comment]
Wow, that is wild. I agree that it sounds like some prototype stuffing via Facebook SDK.

If you can track down _where_ in Assets those extra keys are added, it might give us a clue. If you do `console.log([], Array.prototype, {}, Object.prototype)` and see if that has getParams, getConfig, etc? Can you repro this outside Pixi?

**Comment by talhaozdemir** [comment]
these?

![image](https://github.com/pixijs/pixijs/assets/37840852/f9e081ec-6ed0-419b-9dfa-92b6b1a363f3)

**Comment by talhaozdemir** [comment]
I don't think that I can reproduce this issue outside of Pixi. On the other hand, I run it under the Facebook URL as a localhost. I mean the url is like: 

https://www.facebook.com/embed/instantgames/1111111111111111/player?game_url=https%3A%2F%2Flocalhost%3A8080

**Comment by bigtimebuddy** [comment]
That's super annoying on Facebook's part. 

One idea is that we might want to use `Object.create(null)` instead of `{}` which would remove any prototype-chain and insulate us from this issue. There are a bunch of areas in Assets that do this type of assignment. For example:

https://github.com/pixijs/pixijs/blob/abecc358a0ce73fc301611ca5c2eecfd3dc7c38f/packages/assets/src/Assets.ts#L556

**Comment by SuperSodaSea** [comment]
As a workaround, maybe you can try to `Object.freeze(Object.prototype)` before everything?

**Comment by bigtimebuddy** [comment]
This is the code likely where this is happening:

https://github.com/pixijs/pixijs/blob/abecc358a0ce73fc301611ca5c2eecfd3dc7c38f/packages/assets/src/resolver/Resolver.ts#L651-L656

There are other ways we could write this which would exclude things on the prototype.

**Comment by bigtimebuddy** [comment]
For instance, `Object.getOwnPropertyNames(results)` would give us more stable results.

**Comment by SuperSodaSea** [comment]
I mean, `Object.freeze(Object.prototype)` before `window.fbAsyncInit = function() { ...`

**Comment by SuperSodaSea** [comment]
@bigtimebuddy Perhaps we could consider adding [`guard-for-in`](https://eslint.org/docs/latest/rules/guard-for-in) to our `@pixi/eslint-config`?

<https://github.com/pixijs/eslint-config/blob/0bc65621ec0fc2e477f1c060ceb6a94b8975e5ac/index.js#L66>

**Comment by talhaozdemir** [comment]
@bigtimebuddy I tried your last suggestion:

![image](https://github.com/pixijs/pixijs/assets/37840852/7a612282-745c-45cd-8411-273b67349151)

and I got a different error

![image](https://github.com/pixijs/pixijs/assets/37840852/44593009-19be-4316-9234-015f26702a61)

**Comment by talhaozdemir** [comment]
@SuperSodaSea in your case: `Object.freeze(Object.prototype)` before `window.fbAsyncInit = function() { ...`

 I get TypeErrors:

![image](https://github.com/pixijs/pixijs/assets/37840852/0fd2bb11-3150-4525-9a4c-64590eee2541)

**Comment by SuperSodaSea** [comment]
What about `Object.seal(Object.prototype)` or `Object.preventExtensions(Object.prototype)`?

**Comment by talhaozdemir** [comment]
Nice, both of them solved the issue, I can use these for quick fix.

**Comment by Zyie** [comment]
It's been a long time since this issue was raised, can anyone confirm whether this is still an issue in the latest Facebook SDK?
